jep/301/README.adoc

```diff
@@ -0,0 +1,348 @@
+
+|===
+
+
+== Abstract
+
+A key aspect of
+link:https://github.com/jenkinsci/jep/tree/master/jep/300[Jenkins Essentials]
+is the
+link:https://github.com/jenkinsci/jep/tree/master/jep/300#auto-update[automatically updating distribution],
+which aims to provide an easier to use, self-updating distribution of Jenkins
+core and an "essential" set of plugins. This document outlines the design of
+the Evergreen packaging system for Jenkins Essentials which is, in short, a
+self-updating Docker container necessary to provide the least-error-prone
+mechanism for automatically self-updating Jenkins and a set of plugins.
+
+
+== Specification
+
+[TIP]
+====
+Provide a detailed specification what is being proposed.
+Be as technical and detailed as needed to allow new or existing Jenkins developers
+to reasonably understand the scope/impact of an implementation.
+====
+
+This document proposes the creation of a new Docker image in the
+link:https://hub.docker.com/r/jenkins/[jenkins Docker Hub organization]
+referred to henceforth as `jenkins/evergreen`.
+
+Unlike the current primary images footnoteref:[docker, https://github.com/jenkinsci/docker],
+`jenkins/jenkins:latest`, `jenkins/jenkins:lts`, `jenkins/jenkins:alpine`, and
+`jenkins/jenkins:lts-alpine`, the `jenkins/evergreen` image will contain
+additional scripting, tooling, and machinery to coordinate the
+automatically self-updating functions of Jenkins Essentials.
+
+The first additional tool added into `jenkins/evergreen` is
+link:http://supervisord.org/[supervisord],
+a Python-based process control and supervision daemon. The `supervisord`
+process is responsible for maintaining a proper running state for the two
+subsequent processes which must be launched within the container:
+
+* Jenkins core (`jenkins.war`)
+* <<evergreen-client>>
+
+As described in the diagram below:
+
+[source]
+----
++-------------------------------+
+|  jenkins/evergreen            |
+|-------------------------------|                             |
+| ENTRYPOINT:                   |
+| +---------------------------+ |
+| | supervisord               | |
+| | +-----------------------+ | |
+| | |node: evergreen client | | |
+| | +-----------------------+ | |
+| | +-----------------------+ | |
+| | |java: jenkins.war      | | |
+| | +-----------------------+ | |
+| +---------------------------+ |
++------------------------------ +
+----
+
+`supervisord` is also responsible for reporting process status and allowing
+<<evergreen-client>> to inspect the current running state and information of
+the Jenkins core process.
+
+
+[[evergreen-client]]
+=== Evergreen Client
+
+The second additional tool added into `jenkins/evergreen` is the
+`evergreen-client` process.  In short, it is the process responsible for
+managing updates, interrogating `supervisord` for process status, and reporting
+necessary telemetry to the Evergreen hosted service layer. The
```

Comment:
Should "Evergreen" here should be "Jenkins Essentials"?

Reply:
No, I'm using the name "Evergreen" to describe the distribution mechanism in its entirety which covers client and server-side components